Lib/test/test_warnings/__init__.py

```diff
@@ -877,6 +877,18 @@ def test_showwarning(self):
                                 file_object, expected_file_line)
         self.assertEqual(expect, file_object.getvalue())
 
+        # bpo-35178: Test custom formatwarning can receive line as positional
+        def formatwarning(message, category, filename, lineno, text):
+            return text
+
+        file_object = StringIO()
+        original = self.module.formatwarning
+        self.module.formatwarning = formatwarning
```

Comment:
I would prefer to use "with mock.patch.object(self.module, 'formatwarning', formatwarning): ..." or "with support.swap_attr(self.module, 'formatwarning', formatwarning): ..." here, to ensure that the attribute is restored on error.

Reply:
I initially had `self.addCleanup` to restore it but your suggestion looks more compact and robust. I used `support.swap_attr` since `support` was already imported. Thanks